### PR TITLE
Rework how certificate management happens

### PR DESCRIPTION
**Pull request by hydrogen18**
This refactors the certificate management commands to be non-interactive to the extent that is possible. This should eliminate the chance for user confusion around the workflow. It is now split into 3 steps

1. generate - create the cert locally
2. publish - put the cert on the blockchain
3. revoke - update the blockchain to indicate the cert is no longer valid

None of these should be interactive, except for the prompt to send a transaction. Of note is that the `generate` command is under the `akash tx` command but there is not really anything going on there that is sending a transaction. I can  move this to somewhere else if someone suggests it.

Most of the logic to handle the keys is now wrapped into a single interface.

I haven't changed the location of the files in the home directory or the format of them at this time.

I added a number of unit tests, which try and cover the obvious cases.

### CODE REVIEW

File: x/cert/client/cli/tx_cmd.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+	if err := viper.BindPFlag(flagToGenesis, cmd.Flags().Lookup(flagToGenesis)); err != nil {
+		return err
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return nil
+}
+
+func addCertToGenesis(cmd *cobra.Command, cert types.GenesisCertificate) error {
```
Comment (boz) [inline x/cert/client/cli/tx_cmd.go:148]: I still think that this shouldn't really be needed.

**Review by boz — APPROVED** [review]
Looks great!

**Review by dmikey — APPROVED** [review]

File: _run/common.mk [hunk 2]
```diff
@@ -78,11 +78,13 @@ node-init-genesis-certs: $(patsubst %,node-init-genesis-client-cert-%,$(CLIENT_C
 
 .PHONY: node-init-genesis-client-cert-%
 node-init-genesis-client-cert-%:
-	$(AKASH) tx cert create client --to-genesis=true --from=$(@:node-init-genesis-client-cert-%=%)
+	$(AKASH) tx cert generate client --from=$(@:node-init-genesis-client-cert-%=%)
```
[the remark this answers is not in the document]
Comment (boz) [reply _run/common.mk:81]: Found it - `$*`

```sh
$ cat Makefile
foo-%:
	echo "$*"
$ make foo-bar
echo "bar"
bar
```

File: x/cert/client/cli/tx_cmd.go (hunk 1 above)
Comment (hydrogen18) [reply x/cert/client/cli/tx_cmd.go:148]: It's used in the examples and is convenient whenever I'm setting up local test nets.

File: _run/common.mk (hunk 2 above)
[the remark this answers is not in the document]
Comment (hydrogen18) [reply _run/common.mk:81]: Will give this a try, not very familiar with make but it should work.

[the remark this answers is not in the document]
Comment (boz) [reply _run/common.mk]: sorry I had made #1520 for this

File: x/cert/client/cli/tx_cmd.go (hunk 1 above)
Comment (boz) [reply x/cert/client/cli/tx_cmd.go:148]: it's just that we don't have it for anything else - providers, etc... are all created via the normal mechanism.  but ok, if it's useful then great.

**Pull request merged**